universal-application-tool-0.0.1/test/views/components/TextFormatterTest.java

```diff
@@ -12,17 +12,22 @@
   @Test
   public void urlsRenderCorrectly() {
     ImmutableList<DomContent> content =
-        TextFormatter.createLinksAndEscapeText("hello google.com http://internet.website");
+        TextFormatter.createLinksAndEscapeText(
+            "Hello google.com (http://internet.website), http://mysite.com...!");
```

Comment:
Optional: I would be inclined to keep the existing urlsRenderCorrectly test as is, and add another test called urlsRenderCorrectly_stripsTrailingPunctuation that tests this URL.

This has two pros: (1) We still test for a url without trailing punctuation. (2) It's clearer to the user that we are testing a simple case and then a complex case. Having a urlsRenderCorrectly_stripsTrailingPunctuation test would make it clearer to a reader why we are testing such a complex string ("Hello google.com (http://internet.website), http://mysite.com...!"). When adding a new feature I am generally inclined to add more test coverage, rather than replace existing tests.

Either way, please add bak a test case for a URL without trailing punctuation, however you choose to do it.

Reply:
Sounds reasonable. Updated PR with commit to leave existing test and add an additional covering punctuation stripping.